roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -82,48 +84,47 @@ def roi_locations(self):
     @staticmethod
     def write_segmentation(segmentation_object, save_path):
         save_path = Path(save_path)
-        if save_path.suffix!='hdf5':
-            raise ValueError('filetype to save must be *.hdf5')
-        filename = save_path.parent
-        if segmentation_object.__class__.__name__=='MultiSegmentationExtractor':
+        assert save_path.suffix in ['.hdf5', '.h5'], "'save_path' must be a *.hdf5 or *.h5 file"
+
+        folder_path = save_path.parent
+        file_name = save_path.name
+        if isinstance(segmentation_object, MultiSegmentationExtractor):
             segext_objs = segmentation_object.segmentations
-            if segext_objs.__class__.__name__!='CaimanSegmentationExtractor':
-                raise ValueError('provide a MultisegmentationExtractor of multiple CaimanSegmentationExtractor objects')
             for plane_num, segext_obj in enumerate(segext_objs):
-                save_path_new = save_path.parent.joinpath(f'Plane_{plane_num}').joinpath(filename)
-                CaimanSegmentationExtractor.write_segmentation(segext_obj, save_path_new)
-        if not save_path.parent.exists():
-            save_path.parent.mkdir()
-        else:
-            if save_path.exists():
-                save_path.unlink()
-        with h5py.File(save_path,'a') as f:
-            #create base groups:
+                save_path_plane = folder_path / f'Plane_{plane_num}' / file_name
+                CaimanSegmentationExtractor.write_segmentation(segext_obj, save_path_plane)
+        if not folder_path.is_dir():
+            folder_path.mkdir(parents=True)
+
+        with h5py.File(save_path, 'a') as f:
+            # create base groups:
             estimates = f.create_group('estimates')
             params = f.create_group('params')
-            #adding to estimates:
-            if segmentation_object._roi_response_neuropil is not None:
-                estimates.create_dataset('C',data=segmentation_object._roi_response_neuropil)
-            estimates.create_dataset('F_dff', data=segmentation_object._roi_response_fluorescence)
-            if segmentation_object._roi_response_deconvolved is not None:
-                estimates.create_dataset('S', data=segmentation_object._roi_response_deconvolved)
-            if segmentation_object._image_correlation is not None:
-                estimates.create_dataset('Cn', data=segmentation_object._images_correlation)
+            # adding to estimates:
+            if segmentation_object.get_traces(name='neuropil') is not None:
```

Comment:
in general, it's better to use exposed variables and methods of an object, not private ones